nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/log/kafka/KafkaUtils.java

```diff
@@ -115,34 +90,23 @@ public static Properties getDefaultAdminProperties() {
         return ret;
     }
 
-    public static String getZkServers() {
-        return System.getProperty(ZK_SERVERS_PROP, DEFAULT_ZK_SERVERS);
-    }
-
     public static String getBootstrapServers() {
         return System.getProperty(BOOTSTRAP_SERVERS_PROP, DEFAULT_BOOTSTRAP_SERVERS);
     }
 
     public static boolean kafkaDetected() {
-        return kafkaDetected(getZkServers());
-    }
-
-    public static boolean kafkaDetected(String zkServers) {
+        AdminClient client = null;
```

Comment:
The pattern is to have the init before the `try`:

    AdminClient client = AdminClient.create(getDefaultAdminProperties());
    try {
        ...
    } finally {
        client.close(); // no need to check for null if the factory never returns null
    }

But if `AdminClient` is `AutoCloseable` you can just use try-with-resources.

Reply:
I will fix the pattern but this is the scala AdminClient which is not AutoCLoseable